pkg/cmd/enable_feature_test.go

```diff
@@ -0,0 +1,262 @@
+package cmd_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/api/v1alpha1"
+	"github.com/codeready-toolchain/toolchain-common/pkg/test"
+	"github.com/kubesaw/ksctl/pkg/cmd"
+	clicontext "github.com/kubesaw/ksctl/pkg/context"
+	. "github.com/kubesaw/ksctl/pkg/test"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestEnableFeatureCmd(t *testing.T) {
+	// given
+	SetFileConfig(t, Host())
+	config := configWithFeatures([]toolchainv1alpha1.FeatureToggle{
+		{
+			Name: "feature-x",
+		},
+	})
+
+	var combinations = []struct {
+		alreadyEnabled string
+		afterEnable    map[string]string
+	}{
+		{
+			alreadyEnabled: "",
+			afterEnable: map[string]string{
+				toolchainv1alpha1.FeatureToggleNameAnnotationKey: "feature-x",
+			},
+		},
+		{
+			alreadyEnabled: "feature0",
+			afterEnable: map[string]string{
+				toolchainv1alpha1.FeatureToggleNameAnnotationKey: "feature0,feature-x",
+			},
+		},
+		{
+			alreadyEnabled: "feature1,feature2,feature3",
+			afterEnable: map[string]string{
+				toolchainv1alpha1.FeatureToggleNameAnnotationKey: "feature1,feature2,feature3,feature-x",
+			},
+		},
+	}
+
+	for _, data := range combinations {
+		t.Run("with the already enabled features: "+data.alreadyEnabled, func(t *testing.T) {
+			// given
+			space := newSpace()
+			if data.alreadyEnabled != "" {
+				space.Annotations = map[string]string{
+					toolchainv1alpha1.FeatureToggleNameAnnotationKey: data.alreadyEnabled,
+				}
+			}
```

Comment:
I would keep it as is. Just to make sure we do not touch the annotation map at all if the initial feature list is not set. There is still a difference between an empty map and nil.

Reply:
yes, as Alexey mentioned, it's not the same. We need to make sure that the logic will work when annotations are no initialized (the map is nil). If we don't handle such a case properly, then we could end up with panic trying to modify a map that is nil.

Reply:
I see, but applying my suggestion the tests are still passing, so I'm wondering if we are missing some specific case that sets the annotation to nil to cover this?

Reply:
It's the other way around. Your version puts lower requirements on the code than the current test.
Imagine that we don't have these 3 lines of the code https://github.com/kubesaw/ksctl/blob/b33810a9d13fbac9a5d29087fe6477a06ec992c3/pkg/cmd/enable-feature.go#L95-L97
Your version of the tests will pass, but the current version will fail

Reply:
Ok I see now, as you explained in our 1:1 this check makes so that we test the nil scenario by not setting any annotation in the setup part of the test and leaving it as nil in the first iterartion. Thanks for explaining and sorry fo the confusion.